builder.go

```diff
@@ -332,7 +332,7 @@ func ParseFile(path string) (*parser.Node, error) {
 
 // Step creates a new step from the current state.
 func (b *Builder) Step() *Step {
-	dst := make([]string, len(b.Env)+len(b.RunConfig.Env))
+	dst := make([]string, len(b.Env)+len(b.RunConfig.Env)+len(b.Arguments()))
 	copy(dst, makeUserArgs(b))
 	dst = append(dst, b.RunConfig.Env...)
```

Comment:
Some of the values from `b.Args` won't be used if they conflict with environment values, so the size count being used here for allocating `dst` can be off, and if it is, it can leave a gap with empty strings in the final slice.